src/submodules/pathfinding/astar_continuous.jl

```diff
@@ -0,0 +1,199 @@
+"""
+    find_continuous_path(pathfinder, from, to, model)
+Functions like `find_path``, but uses the output of `find_path` and converts it to the coordinate
+space used by `model.space` (which is continuous). Also performs checks on the last two waypoints
+in the discrete path to ensure continuous path is optimal.
+"""
+function find_continuous_path(
+    pathfinder::AStar{D},
+    from::NTuple{D,Float64},
+    to::NTuple{D,Float64},
+    model::ABM{<:ContinuousSpace{D}}
+) where {D}
+    # used to offset positions, so edge cases get handled properly (i.e. (0., 0.) maps to grid
+    # cell (1, 1))
+    half_cell_size = model.space.extent ./ pathfinder.grid_dims ./ 2.
+    discrete_from = Tuple(Agents.get_spatial_index(from, pathfinder.walkable, model))
+    discrete_to = Tuple(Agents.get_spatial_index(to, pathfinder.walkable, model))
+    discrete_path = find_path(pathfinder, discrete_from, discrete_to)
+    isnothing(discrete_path) && return
+    isempty(discrete_path) && return Path{D,Float64}()
+    cts_path = Path{D,Float64}()
```

Comment:
It'd be good to have a few comments in the code here to explain this logic. I know its all edge cases, but this kind of stuff is difficult to reason with six months from now when we forget about it.